app/components/request/raw-config.js

```diff
@@ -15,52 +16,46 @@ export default Component.extend({
   copied: false,
   baseYmlName: '.travis.yml',
 
-  isExpanded: computed('rawConfig.config', function () {
-    return this.get('rawConfig.config') !== '{}';
-  }),
+  isEmptyConfig: equal('config', '{}'),
+  isExpanded: not('isEmptyConfig'),
 
   toggleStatusClass: computed('isExpanded', function () {
     return this.isExpanded ? 'expanded' : 'collapsed';
   }),
 
-  buttonLabel: computed('copied', 'rawConfig.source', function () {
-    let source = this.get('rawConfig.source');
-    return this.copied ? 'Copied!' : `Copy ${fileNameWithoutSha(source)}`;
+  buttonLabel: computed('copied', 'source', function () {
+    return this.copied ? 'Copied!' : `Copy ${fileNameWithoutSha(this.source)}`;
   }),
 
-  formattedConfig: computed('rawConfig.config', 'slug', function () {
-    let config = this.get('rawConfig.config');
+  formattedConfig: computed('config', 'slug', function () {
     try {
-      return JSON.stringify(JSON.parse(config), null, 2);
+      return JSON.stringify(JSON.parse(this.config), null, 2);
     } catch (e) {
-      return config;
+      return this.config;
     }
```

Comment:
nice, i've tried this out! however, the logic doesn't seem to be precisely the same, and it would incorrectly re-stringify a YAML string that JSON.parse has failed parsing: the current logic does not `JSON.stringify` the string if `JSON.parse` failed. it only does so if `JSON.parse` succeeded.

we can still get rid of the try/catch statement though:

```
    let object = parseWithDefault(this.config, null);
    return object ? JSON.stringify(object, null, 2) : this.config;
```

if the ternary is acceptable in this codebase then i think this is a huge improvement (i personally love ternaries ;) ...  without it we'd need an if statement, and i'm not sure if that's then more expressive or readable than just keeping the try/catch statement instead.

i have committed this for now: https://github.com/travis-ci/travis-web/commit/824283207d347de2ad65beea1c97e2392a811b9e